lms/djangoapps/class_dashboard/dashboard_data.py

```diff
@@ -180,11 +187,29 @@ def get_d3_problem_grade_distrib(course_id):
                         # Construct label to display for this problem
                         label = "P{0}.{1}.{2}".format(c_subsection, c_unit, c_problem)
 
-                        # Only problems in prob_grade_distrib have had a student submission.
-                        if child.location in prob_grade_distrib:
+                        grade_distribution = []
+
+                        try:
+                            problem = client.modules(course_id, child.location)
+                            grade_distribution = problem.grade_distribution()
+                        except NotFoundError:
+                            pass
+
+                        # Only problems with recorded scores have had student submissions.
+                        if len(grade_distribution) > 0:
```

Comment:
I'm going to stick with the > 0, since the conditional's purpose here isn't immediately apparent, and it makes the comment on L198 a little easier to understand.